Cleanup

andreGarvin opened the pull request:
I remove the node_modules folder and add .gitgnore so that does not happen again

Commit:
- removed node_modules
- add .gitignore file
- cleanup server.js file

Commit:
whoops

osamahan999 commented:
Could you point out what changes you made in terms of additions? I see +18 additions but I cannot find where these additions are made, other than .gitignore.

andreGarvin commented:
sorry if the message was not that clear. I added a .gitignore file and added not node_modules so that it would not be committed to history next time you make another commit. After that ai cleaned up the server.js file for most of the imports and middleware and routes. That was pretty much it, the reason you have that many files are because I deleted the node_modules from history

andreGarvin commented:
Also, you might want lock down those package versions

osamahan999 commented:
Doesnt package-lock.json lock the package versions and the dependencies? Perhaps I am missing something, apologies, I am very new to the npm ecosystem

andreGarvin commented:
when you have `^` in front of a version is means this version or the latest version. This can lead to accidentally using the latest version of the package instead of the version you installed. Plus it helps know that it is only that version that is being used.

Don't have to apologize for not knowing it, everyone has to start from somewhere when learning. feel free to contact me in Twitter if you want to ask any more questions

osamahan999 commented:
Sent you a follow request :)! I removed the ^ from the package.json. I assume I do not need to do that in my package-lock file
